FrostyPlugin/Controls/Editors/FrostyNumberEditor.cs

```diff
@@ -99,22 +125,63 @@ public override ValidationResult Validate(object value, CultureInfo cultureInfo)
                     else if (type == typeof(ushort)) ushort.Parse(strValue);
                     else if (type == typeof(int))
                     {
-                        try { int.Parse(strValue); }
+                        try
+                        {
+                            if (int.TryParse(strValue, out int _) || TrySolve(strValue, out double _))
+                            {
+                                return new ValidationResult(true, null);
+                            }
+                            else
+                            {
+                                return new ValidationResult(false, null);
+                            }
+                        }
                         catch (OverflowException) { return new ValidationResult(false, null); }
                         catch (Exception) { }
                         return new ValidationResult(true, null);
                     }
                     else if (type == typeof(uint))
                     {
-                        try { uint.Parse(strValue); }
+                        try
+                        {
+                            if (uint.TryParse(strValue, out uint _) || TrySolve(strValue, out double _))
+                            {
+                                return new ValidationResult(true, null);
+                            }
+                            else
+                            {
+                                return new ValidationResult(false, null);
+                            }
+                        }
                         catch (OverflowException) { return new ValidationResult(false, null); }
```

Comment:
those are also not needed if u use the tryparse right?

Reply:
yea theyre not needed